backend/api/view/AuthorProfileView.py

```diff
@@ -50,29 +57,90 @@ def post(self, request, uid):
 
     def get(self, request, uid):
         authorId = self.kwargs['uid']
-        if(authorId == ""):
+        if (authorId == ""):
             return Response("Error: Author ID required!", status.HTTP_400_BAD_REQUEST)
 
-        query_set = AuthorProfile.objects.filter(id=authorId)
+        author_profile_exists = AuthorProfile.objects.filter(user=request.user).exists()
+        server_user_exists = ServerUser.objects.filter(user=request.user).exists()
 
-        if (len(query_set) == 1):
-            response_data = AuthorProfileSerializer(query_set[0]).data
-            friends = Follow.objects.filter(authorA=response_data["id"], status="FRIENDS")
-            friends_list_data = []
-            for ele in friends:
-                friend_fulll_id = ele.authorB
-                tmp = friend_fulll_id.split("author/")
-                host = tmp[0]
-                short_id = tmp[1]
-                # todo: check if host belongs to our server, call cross server endpoint if doesnt
-                friend_profile = AuthorProfile.objects.get(id=short_id)
-                serialized_author_profile = AuthorProfileSerializer(friend_profile)
+        # from front end
+        if author_profile_exists:
+            is_local_uuid = True
+            # for foreign author:
+            # expect front end to send http://127.0.0.1:8000/author/http%3A%2F%2F127.0.0.1%3A1234%2Fauthor%2F163974c0-b350-4e9b-a708-b570acee826d
+            # for local author:
+            # expect to front end to send http://127.0.0.1:8000/author/163974c0-b350-4e9b-a708-b570acee826d
+            try:
+                uuid.UUID(uid)
+            except ValueError:
+                is_local_uuid = False
+            if is_local_uuid:
+                try:
+                    author_profile = AuthorProfile.objects.get(id=authorId)
+                    response_data = AuthorProfileSerializer(author_profile).data
+                    friends = Follow.objects.filter(authorA=response_data["id"], status="FRIENDS")
+                    friends_list_data = []
+                    for ele in friends:
```

Comment:
not too picky about this but i like for friend in friends better becasue it is more meaningful